c++/src/capnp/any.h

```diff
@@ -806,6 +806,31 @@ inline Orphan<AnyPointer> Orphan<AnyPointer>::releaseAs() {
   return kj::mv(*this);
 }
 
+enum class StructEqualityResult {
+  NOT_EQUAL,
+  EQUAL,
+  UNKNOWN_CONTAINS_CAPS
+};
+
+inline kj::StringPtr KJ_STRINGIFY(StructEqualityResult res) {
+  switch(res) {
+  case StructEqualityResult::NOT_EQUAL:
+    return "NOT_EQUAL";
+  case StructEqualityResult::EQUAL:
+    return "EQUAL";
+  case StructEqualityResult::UNKNOWN_CONTAINS_CAPS:
+    return "UNKNOWN_CONTAINS_CAPS";
+  }
+}
+
+StructEqualityResult equal(AnyStruct::Reader left, AnyStruct::Reader right);
```

Comment:
I'd prefer to see these defined as methods, both for namespacing reasons and so that people can find them when reading the AnyStruct etc. class definitions.